### PR TITLE
add annotations and labels when managed cluster exists

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>

# Description of the change(s):
* when the managed cluster exists, we add labels and annotations for it.

## Why do we need this PR:
*  This is good for the multi-hub scenarios, in multiple hub scenario, we want to create managed cluster first, then create the hypershiftDeployment. with this PR, the hypershift deployment controller can ensure these labels and annotations.

## Issue reference: 
* 

## Test API/Unit - Success
```script
?   	github.com/stolostron/hypershift-deployment-controller/api/v1alpha1	[no test files]
?   	github.com/stolostron/hypershift-deployment-controller/pkg	[no test files]
ok  	github.com/stolostron/hypershift-deployment-controller/pkg/client	2.035s	coverage: 87.5% of statements
?   	github.com/stolostron/hypershift-deployment-controller/pkg/constant	[no test files]
ok  	github.com/stolostron/hypershift-deployment-controller/pkg/controllers	39.854s	coverage: 80.8% of statements
ok  	github.com/stolostron/hypershift-deployment-controller/pkg/controllers/autoimport	4.527s	coverage: 62.1% of statements
ok  	github.com/stolostron/hypershift-deployment-controller/pkg/helper	3.180s	coverage: 62.5% of statements
ok  	github.com/stolostron/hypershift-deployment-controller/test/integration	41.447s	coverage: [no statements]
```

### CODE REVIEW

**Comment by zhujian7** [comment]
cc @rokej @mikeshng @elgnay

File: pkg/controllers/autoimport/autoimport_controller.go [hunk 1]
```diff
@@ -195,57 +195,107 @@ func ensureManagedCluster(r *Reconciler, hydNamespaceName types.NamespacedName,
...
+		if setAnnotationIfNotPresent(mc, k, v) {
+			annoChanged = true
+		}
+	}
+
+	return labelChanged || annoChanged
+}
+
+func setLabelIfNotPresent(mc *mcv1.ManagedCluster, key, value string) bool {
+	if v, ok := mc.Labels[key]; !ok || len(v) == 0 {
```
Comment (elgnay) [inline pkg/controllers/autoimport/autoimport_controller.go:284]: `len(v) == 0` => `v != value`?

File: pkg/controllers/autoimport/autoimport_controller.go [hunk 2]
```diff
@@ -195,57 +195,107 @@ func ensureManagedCluster(r *Reconciler, hydNamespaceName types.NamespacedName,
...
+		mc.Labels[key] = value
+		return true
+	}
+	return false
+}
+
+func setAnnotationIfNotPresent(mc *mcv1.ManagedCluster, key, value string) bool {
+	if v, ok := mc.Annotations[key]; !ok || len(v) == 0 {
```
Comment (elgnay) [inline pkg/controllers/autoimport/autoimport_controller.go:292]: `len(v) == 0` => `v != value?`

File: pkg/controllers/autoimport/autoimport_controller.go (hunk 1 above)
Comment (elgnay) [reply pkg/controllers/autoimport/autoimport_controller.go:284]: That's fine if it is allowed to modify the values of those labels, like the clusterset label.

Comment (zhujian7) [reply pkg/controllers/autoimport/autoimport_controller.go:284]: I think we do not need to update the value, we just set it when it is empty or not present.

**Review by rokej — APPROVED** [review]

**Pull request merged**